### PR TITLE
fix: Fixes JSON parsing issue #748

### PR DESCRIPTION
**Pull request by orlenko**

### CODE REVIEW

**Comment by orlenko** [comment]
Hi! Just checking - can someone review the fix please?

**Comment by orlenko** [comment]
Hello, this is your monthly nag about a pending bugfix. Please review!

File: src/transports/websocket.ts [hunk 1]
```diff
@@ -67,10 +61,40 @@ export default class WebSocketServerTransport extends ServerTransport {
...
+  private onMessage(ws: WebSocket, message: string) {
+    try {
+      return this.webSocketRouterHandler(
+        JSON.parse(message),
+        ws.send.bind(ws)
+      );
+    } catch (e) {
+      ws.send(
+        JSON.stringify({
+          id: 0,
```
Comment (zcstarr) [inline src/transports/websocket.ts:82]: I think this in theory should be null.

Comment (zcstarr) [inline src/transports/websocket.ts:82]:
```suggestion
          id: null,
```

**Review by zcstarr — COMMENTED** [review]
Overall I think this looks great! I added a comment about the null. I want to have 1 more look next week just to make sure I don't have late Friday eyes. But thank you for the awesome PR let's try and get this landed 👍

File: src/transports/websocket.ts [hunk 2]
```diff
@@ -67,10 +61,40 @@ export default class WebSocketServerTransport extends ServerTransport {
...
+        req.map((r: JSONRPCRequest) => super.routerHandler(r))
+      );
     } else {
       result = await super.routerHandler(req);
     }
     respondWith(JSON.stringify(result));
   }
+
+  private onMessage(ws: WebSocket, message: string) {
+    try {
```
Comment (BelfordZ) [inline src/transports/websocket.ts:74]: can we first try to parse the JSON, and fail early (before calling router handler method).
aka
```
let jsonMsg;
try {
  jsonMsg = JSON.parse(message);
} catch (e) {
  ... ws.send ...
}
```

Comment (orlenko) [reply src/transports/websocket.ts:75]: Addressed in 7d3a561 — please have a look

File: src/transports/websocket.ts (hunk 1 above)
Comment (orlenko) [reply src/transports/websocket.ts]: Thanks @zcstarr!

**Review by zcstarr — APPROVED** [review]

**Comment by zcstarr** [comment]
@orlenko This looks good overall thanks! for making the changes. If @BelfordZ doesn't have a look by 5/10, squash it down to a single commit and make the comment semantic `fix: fixes JSON parsing issue #748 ` I'll merge it in.

**Comment by zcstarr** [comment]
@orlenko please squash it down with the following commit miessage `fix: invalid JSON parsing issue fixes #748` and the we can get this squashed and merged in

**Comment by bakon11** [comment]
This is exactly what I was looking for, thank you for getting it done, when squash and merge? :grimacing:

**Comment by orlenko** [comment]
Sorry about the delay, squashing now...

**Comment by orlenko** [comment]
oh geez, I messed up with the squash/merge, sorry. Can someone please give me a hint how to do it properly, before I make things worse? Or squash it if that's easier. So sorry about the mess!

**Comment by orlenko** [comment]
@zcstarr ^^ My apologies

**Pull request closed without merging**